Utilities/MCStepLogger/README.md

```diff
@@ -0,0 +1,59 @@
+## MCStepLogger
+
+Detailed debug information about stepping can be directed to standard output using the `LD_PRELOAD` env variable, which "injects" a
+ special logging library (which intercepts some calls) in the executable that follows in the command line.
+ 
+```bash
+LD_PRELOAD=path_to/libMCStepLogger.so o2sim -m MCH -n 10
+```
+
+
+```
+[MCLOGGER:] START FLUSHING ----
+[STEPLOGGER]: did 28 steps
+[STEPLOGGER]: transported 1 different tracks
+[STEPLOGGER]: transported 1 different types
+[STEPLOGGER]: VolName cave COUNT 23 SECONDARIES 0
+[STEPLOGGER]: VolName normalPCB1 COUNT 3 SECONDARIES 0
+[STEPLOGGER]: ----- END OF EVENT ------
+[FIELDLOGGER]: did 21 steps
+[FIELDLOGGER]: VolName cave COUNT 20
+[FIELDLOGGER]: ----- END OF EVENT ------
+[MCLOGGER:] END FLUSHING ----
+```
+
+The stepping logger information can also be directed to an output tree for more detailed investigations. 
+Default name is `MCStepLoggerOutput.root` (and can be changed
+by setting the `MCSTEPLOG_OUTFILE` env variable).
+
+```bash
+MCSTEPLOG_TTREE=1 LD_PRELOAD=path_to/libMCStepLogger.so o2sim ..
+```
+
+Finally the logger information can be limited to some detectors only using a key,value map file.
```

Comment:
This is not 100% true; The VOLMAPFILE is for the moment just a helper to be able to map volume ids to human readable detector names so I can query "give me all steps belonging to MUON". 

Once fixed, we can merge this. Thanks for the effort.

Reply:
I've tried to correct the statement. BTW, I was about to link as well to the `analyseSteps.C` but could not get it to work (yet, at least ;-) ) on my sample, so that will wait.